tools/Customization/DevHome.FileExplorerSourceControlIntegration/Services/RepositoryTracking.cs

```diff
@@ -150,6 +150,23 @@ public string GetSourceControlProviderForRootPath(string rootPath)
         }
     }
 
+    public void ModifySourceControlProviderForTrackedRepository(string extensionCLSID, string rootPath)
+    {
+        lock (trackRepoLock)
+        {
+            if (TrackedRepositories.ContainsKey(rootPath))
+            {
+                TrackedRepositories[rootPath] = extensionCLSID;

```

Comment:
I am confused by this method.  What is being modified?  Are you re-assigning the repo to a different extension?

Reply:
Yes, the source control provider for the repository is being changed. I updated the logging to reflect this as well. Thanks!